src/plugins/zoom.ts

```diff
@@ -59,20 +75,33 @@ class ZoomPlugin extends BasePlugin<ZoomPluginEvents, ZoomPluginOptions> {
     // prevent scrolling the sidebar while zooming
     e.preventDefault()
 
-    const duration = this.wavesurfer.getDuration()
-    const oldMinPxPerSec = this.wavesurfer.options.minPxPerSec
-    const x = e.clientX
-    const width = this.container.clientWidth
-    const scrollX = this.wavesurfer.getScroll()
-    const pointerTime = (scrollX + x) / oldMinPxPerSec
-    const newMinPxPerSec = this.calculateNewZoom(oldMinPxPerSec, -e.deltaY)
-    const newLeftSec = (width / newMinPxPerSec) * (x / width)
-    if (newMinPxPerSec * duration < width) {
-      this.wavesurfer.zoom(width / duration)
-      this.container.scrollLeft = 0
-    } else {
-      this.wavesurfer.zoom(newMinPxPerSec)
-      this.container.scrollLeft = (pointerTime - newLeftSec) * newMinPxPerSec
+    // Update the accumulated delta...
+    this.accumulatedDelta += -e.deltaY
+
+    // ...and only scroll once we've hit our threshold
+    if (this.options.deltaThreshold === 0 || Math.abs(this.accumulatedDelta) >= this.options.deltaThreshold) {
+      const duration = this.wavesurfer.getDuration()
+      const oldMinPxPerSec = this.wavesurfer.options.minPxPerSec
+      const x = e.clientX
+      const width = this.container.clientWidth
+      const scrollX = this.wavesurfer.getScroll()
+      const pointerTime = (scrollX + x) / oldMinPxPerSec
+      const newMinPxPerSec = this.calculateNewZoom(
+        oldMinPxPerSec,
+        this.options.deltaThreshold > 0 ? this.accumulatedDelta : -e.deltaY,
+      )
```

Comment:
I think this can be simplified to 
```suggestion
      const newMinPxPerSec = this.calculateNewZoom(oldMinPxPerSec, this.accumulatedDelta)
```

If the threshold is 0, `this.accumulatedDelta` would be just equal to the delta because it's reset every time.